Shared/Samples/Edit geometries with programmatic reticle tool/EditGeometriesWithProgrammaticReticleToolView.swift

```diff
@@ -0,0 +1,617 @@
+        geometryOverlay.addGraphics([
+            pinkneysGreenGraphic,
+            beechLodgeBoundaryGraphic,
+            treeMarkers
+        ])
+        
+        geometryEditor.tool = reticleTool
+    }
+    
+    /// Saves the current geometry to the graphics overlay and stops editing.
+    /// - Precondition: Geometry's sketch must be valid.
+    func save() {
+        precondition(geometryEditor.geometry?.sketchIsValid ?? false)
+        
+        if selectedGraphic != nil {
+            // Update geometry for edited graphic.
+            updateGraphic()
+        } else {
+            // Add new graphic.
+            addGraphic()
+        }
+        
+        reticleState = .default
+    }
+    
+    /// Updates the selected graphic with the current geometry.
+    private func updateGraphic() {
+        guard let selectedGraphic = selectedGraphic.take() else { return }
+        selectedGraphic.geometry = geometryEditor.stop()
+        isStarted = false
+        selectedGraphic.isVisible = true
+    }
+    
+    /// Adds a new graphic for the current geometry to the graphics overlay.
+    private func addGraphic() {
+        let geometry = geometryEditor.geometry!
+        let graphic = Graphic(geometry: geometry, symbol: symbol(for: geometry))
+        geometryOverlay.addGraphic(graphic)
+        stop()
+        canClearGraphics = true
+    }
+    
+    /// Removes the initial graphics and saved sketches on the graphics overlay.
+    func deleteAllGeometries() {
+        geometryOverlay.removeAllGraphics()
+        canClearGraphics = false
+        reticleState = .default
+    }
+    
+    /// Starts editing with the specified tool and geometry type.
+    /// - Parameter geometryType: The type of geometry to draw.
+    func startEditing(withType geometryType: Geometry.Type) {
+        geometryEditor.start(withType: geometryType)
+        isStarted = true
+        reticleState = .default
+    }
+    
+    /// Starts editing a given graphic with the geometry editor.
+    /// - Parameter graphic: The graphic to edit.
+    func startEditing(with graphic: Graphic) {
+        selectedGraphic = graphic
+        graphic.isVisible = false
+        let geometry = graphic.geometry!
+        geometryEditor.start(withInitial: geometry)
+        isStarted = true
+        updateReticleState()
+    }
+    
+    /// Stops editing with the geometry editor.
+    func stop() {
+        geometryEditor.stop()
+        isStarted = false
+        if let selectedGraphic = selectedGraphic.take() {
+            selectedGraphic.isVisible = true
+        }
+        reticleState = .default
+    }
+    
+    /// Returns the symbology for graphics saved to the graphics overlay.
+    /// - Parameter geometry: The geometry of the graphic to be saved.
+    /// - Returns: Either a marker or fill symbol depending on the type of provided geometry.
+    private func symbol(for geometry: Geometry) -> Symbol {
+        switch geometry {
```

Comment:
We typically don't omit the `return` keyword for methods that aren't all on a single line.

```suggestion
        return switch geometry {
```